src/components/formProducts/FormProducts.js

```diff
@@ -2,12 +2,23 @@ import React, { useRef, useState } from 'react';
 import { Outlet } from 'react-router-dom';
 import { getTokenFromStorage } from '../../utils/utils';
 import { ProductForList } from '../ProductForList';
+import './FormProducts.css';
 
 const FormProducts = ({ items }) => {
   const [itemName, setItemName] = useState('');
   const handleChange = (e) => setItemName(e.target.value);
   const token = useRef(getTokenFromStorage());
 
+  items = !itemName
+    ? items
+    : items.filter((filterItem) =>
```

Comment:
Estoy realmente, sorprendido como estas lineas.... wow!!! es magía pura... sin embargo... como sugerencia, consideren usar estados y `useEffect`

```suggestion
  const [itemsFiltered, setItemsFiltered] = useState(items);
  //... more code
    useEffect(() => {
    if (itemName === '') {
      setItemsFiltered(items)
    } else {
      setItemsFiltered(items.filter((filterItem) =>
        filterItem.name.toLowerCase().includes(itemName.toLocaleLowerCase())))
    }
  }, [itemName, items])
  //... more code
  {itemsFiltered && // more code}
```

Reply:
El filtro funciona súper pero creo que sí sería mejor usar estados por el siguiente detalle: cuando se marca un producto como comprado e inmediatamente se busca con el filtro éste aparece desmarcado.

Creo que esto pasa porque el filtro trabaja sobre el `items` inicial (donde el producto no estaba marcado). No es sino hasta que vuelve a haber un render de `ListProducts` que las marcas se reflejan en el filtro.